src/lsp/cobol_lsp/lsp_request.ml

```diff
@@ -454,6 +454,61 @@ let handle_semtoks_full,
 
 (** {3 Hover} *)
 
+let get_hover_text (qn: Cobol_ptree.qualname) (cu: Cobol_unit.Types.cobol_unit) =
+  match Cobol_unit.Qualmap.find qn cu.unit_data.data_items.named with
+  | Data_field { def; _} as data_def ->
+      Some (Pretty.to_string "%a" Lsp_data_info_printer.pp_data_definition data_def),
+      Some (~@def)
+  | Data_renaming { def; _} as data_def ->
+      Some (Pretty.to_string "%a" Lsp_data_info_printer.pp_data_definition data_def),
+      Some (~@def)
+  | Data_condition { def; _} as data_def ->
+      Some (Pretty.to_string "%a" Lsp_data_info_printer.pp_data_definition data_def),
+      Some (~@def)
+  | Table_index { table; _ } as data_def ->
+      Some (Pretty.to_string "%a" Lsp_data_info_printer.pp_data_definition data_def),
+      Some (~@table)
+  | exception Cobol_unit.Qualmap.Ambiguous _ ->
+      None,None
+
+
+let always_show_hover_text_in_data_div = true
+
+let filter_data_div text def_loc pos filename =
+  let is_not_in_hovered_def = not @@ Lsp_position.is_in_srcloc ~filename pos def_loc in
+  if is_not_in_hovered_def || always_show_hover_text_in_data_div
```

Comment:
Also: please favor `not (…)` to `not @@ …`. (I usually try to avoid parentheses using `@@` myself, but for the case of `not` I think it makes the code more obscure).